M2/cmake/check-libraries.cmake

```diff
@@ -326,9 +326,6 @@ if(FACTORY_FOUND)
   # whether factory was built with --enable-streamio
   check_cxx_source_compiles([[#include <factory/factory.h>
     int main(){Variable x; x = Variable(); std::cout << x;return 0;}]] FACTORY_STREAMIO)
-  # whether Prem() from factory is public
-  check_cxx_source_compiles([[#include <factory/factory.h>
-    int main(){CanonicalForm p,q; Prem(p,q);return 0;}]] HAVE_FACTORY_PREM)
```

Comment:
@mahrud Why did we/you remove this check?  Is this part of something larger, or a change in factory?

Reply:
I see the same change was made for autoconf build

Reply:
@d-torrance submitted https://github.com/Singular/Singular/pull/755 to make Prem public in Factory >4 years ago, but now that I'm looking at the minimum version that we check for, we might still need this. What do you think Doug?

Reply:
It looks like that PR became part of singular 4.1.1, but 4.0.3 is still in some older, but still supported releases of distros, e.g., Debin Stretch (EOL 2022) and Ubuntu Xenial (EOL 2024).  So I'd recommend keeping it around a while longer.

Reply:
Okay, reverted in #1667